File: src/main/java/com/example/money_management/entity/ChatMessage.java
package com.example.money_management.entity;

import com.example.money_management.dto.MemberDTO;
import jakarta.persistence.*;
import lombok.*;
import lombok.extern.log4j.Log4j2;

import java.util.Date;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
@ToString
@Builder
public class ChatMessage {

    @Id
    private Long msg_id;

    @JoinColumn(name = "sender")
    @ManyToOne
    private Member sender;

    @JoinColumn(name = "receiver")
    @ManyToOne
    private Room receiver;

    @Temporal(TemporalType.TIMESTAMP)
    private Date createTime;

    private String msg;
    private String type;

    @PrePersist
    private void createMsgId(){

    }
}

File: src/main/resources/static/js/chat/chatMessage.js
class Message
{
    #msg_id;
    #sender;
    #receiver;
    #msg;
    #type;
    #isMine;
    #create_date;

    constructor(msg_id, sender, receiver, msg, type, isMine, create_date)
    {
        this.#msg_id      = msg_id;
        this.#sender      = sender;
        this.#receiver    = receiver;
        this.#msg         = msg;
        this.#type        = type;
        this.#isMine      = isMine;
        this.#create_date = create_date;
    }

    get isMine()
    {
        return this.#isMine;
    }
    get msg_id()
    {
        return this.#msg_id;
    }
    get sender()
    {
        return this.#sender;
    }
    get receiver()
    {
        return this.#receiver;
    }
    get msg()
    {
        return this.#msg;
    }
    get type()
    {
        return this.#type;
    }
    get createDate()
    {
        return this.#create_date;
    }
    
}
